### PR TITLE
Feat(eos_cli_config_gen): Support Aboot password

### PR DESCRIPTION
**Pull request by tgodaA**
## Change Summary

Support command to set device Aboot password

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes

```yaml
boot:
  # Set the Aboot password
  secret:
    hash_algorithm: < md5 | sha512 | default -> sha512 >
    key: "< hashed_password >"
```

## How to test

Molecule

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md [hunk 1]
```diff
@@ -3344,6 +3345,18 @@ spanning_tree:
...
+### System Boot Settings
+
+```yaml
+boot:
+  # Set the Aboot password
+  secret:
+    # encryption_type 0, LINE -> clear_text_password (UNENCRYPTED)
+    # encryption_type 5, sha512 -> encrypted_password_string (ENCRYPTED)
+    encryption_type: < "0" | "5" | "LINE" | "sha512" >
+    password: < encrypted_password_string | clear_text_password >
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: please remove `encryption_type`
please change `password:` to `sha512_password: "< sha_512_password >"`

**Review by carlbuchmann — CHANGES_REQUESTED** [review]

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md (hunk 1 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: I don't agree. If we want to support multiple types, we will have multiple vars (ex. `clear_text_password`, `md5_password` and `sha512_password`), but only one of them works at a time. But we could have a default of `sha512` if that is the usual one, and then people only have to worry about password.

`LINE` is not a valid option. It is just the EOS Cli help text representation of the actual password.

Comment (tgodaA) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: Will update.

Comment (tgodaA) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: I will add a hierarchy. Check sha512 then md5 then clear_text password.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/boot.j2 [hunk 2]
```diff
@@ -0,0 +1,27 @@
+{# System Boot Configuration #}
+{% if boot is arista.avd.defined %}
+
+# System Boot Settings
+{%     if boot.secret is arista.avd.defined %}
+
+## Boot Secret Summary
+{%         if boot.secret.sha512_password is arista.avd.defined %}
+{%             set method = "SHA512 ENCRYPTED" %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/boot.j2]: SHA512, MD5 are not encryption, but Hash methods/algorithms.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/boot.j2 [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+{%             set password = boot.secret.sha512_password %}
+{%         elif boot.secret.md5_password is arista.avd.defined %}
+{%             set method = "MD5 ENCRYPTED" %}
+{%             set password = boot.secret.md5_password %}
+{%         elif boot.secret.clear_text_password is arista.avd.defined %}
+{%             set method = "UNENCRYPTED" %}
+{%             set password = boot.secret.clear_text_password %}
+{%         endif %}
+
+- The {{ method }} Aboot password is: {{ password }}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/boot.j2]: Let's not print out passwords in documentation. Just say that it is set, similar to enable password.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/boot.j2 [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{# eos - System Boot Configuration #}
+{% if boot is arista.avd.defined %}
+!
+{%     if boot.secret.sha512_password is arista.avd.defined %}
+! boot secret sha512 {{ boot.secret.sha512_password }}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/boot.j2]: `!` first here??

Comment (tgodaA) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/boot.j2]: Yes, that's what I saw on eos device when I ran ```show run```

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/boot.j2]: Yes, but that should not be our output. The command is simply ignores if you paste it like this. In my tests, I had to remove the !

Comment (tgodaA) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/boot.j2]: Ok, got it.

Comment (tgodaA) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/boot.j2]: Correct, I tested it now. Got it what you were saying.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md [hunk 5]
```diff
@@ -3430,6 +3431,18 @@ spanning_tree:
       priority: < priority >
 ```
 
+### System Boot Settings
+
+```yaml
+boot:
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md:3439]: As discussed offline, we should adjust this to:
```yaml
boot:
  # Set the Aboot password
  secret:
    hash_algorithm: < md5 | sha512 | default -> sha512 >
    key: "< hashed_password >"
```

Comment (tgodaA) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md:3439]: Sure, will do.

**Review by ClausHolbechArista — APPROVED** [review]
LGTM

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/DC1_FABRIC.yml [hunk 6]
```diff
@@ -254,3 +254,8 @@ platform_speed_groups:
   7280R:             # Only setting speed-groups on 7280R platform, so only L3leaf should get this setting.
     25G: [ 3, 2 ]    # Unsorted order, but we should sort output correctly.
     10G: [ 1, 2, 4 ] # Duplicate speed-group 2. Since we sort on key first the result will be 25G for group 2
+
```
Comment (emilarista) [inline ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/DC1_FABRIC.yml:257]: Do we really need to set this on all these devices? It's easier to review if we keep the scope smaller, so maybe one or 2 devices is enough.

**Review by emilarista — DISMISSED** [review]

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/DC1_FABRIC.yml (hunk 6 above)
Comment (tgodaA) [reply ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/DC1_FABRIC.yml:257]: Yes to test on all devices.

**Review by carlbuchmann — APPROVED** [review]
LGTM!

**Pull request merged**